File: Lesson 2 - Arrays/OddOccurrencesInArray.cpp
#include <assert.h>
#include <vector>

int solution(std::vector<int> &A)
{
    assert(A.size() % 2);
    int res = 0;
    for (const auto& elem : A)
    {
        res ^= elem;
    }
    return res;
}

int main()
{
    assert(1 == solution(std::vector<int>({ 1 })));
    assert(1 == solution(std::vector<int>({ 1, 1, 1 })));
    assert(3 == solution(std::vector<int>({ 3, 1, 1 })));
    assert(INT_MAX == solution(std::vector<int>({ 1, INT_MAX, 1 })));
    assert(INT_MIN == solution(std::vector<int>({ 1, 1, INT_MIN })));
    assert(7 == solution(std::vector<int>({ 9, 3, 9, 3, 9, 7, 9 })));
}

File: Lesson 2 - Arrays/OddOccurrencesInArray.py
import sys

def solution(A):
    assert len(A) % 2
    res = 0
    for elem in A:
        res ^= elem
    return res

assert 1 == solution([ 1 ])
assert 1 == solution([ 1, 1, 1 ])
assert 3 == solution([ 3, 1, 1 ])
assert sys.maxint == solution([ 1, sys.maxint, 1 ])
assert -sys.maxint - 1 == solution([ 1, 1, -sys.maxint - 1 ])
assert 7 == solution([ 9, 3, 9, 3, 9, 7, 9 ])
